### ISSUE TITLE
Redundant to_a after each_slice goes unnoticed

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#14640]
Prior yields same as latter, but gets no breathalyzer. Not sure if bug or a feature request.

## Steps to reproduce the problem

```ruby
pp 4.times.each_slice(2).to_a.to_h
pp 4.times.each_slice(2).to_h
```

## Expected behavior

"`to_h` is compatible with `each_slice`, so `to_a` is redundant and can be safely removed."

## Actual behavior

Zips it.

## RuboCop version

```
System Version: macOS 26.0.1 (25A362)
Model Identifier: MacBookAir10,1
ruby      2025-08-01 30a20bc16
rubocop   2025-10-18 f7f653f50

```

**Comment by forthrin** [comment]
```ruby
[2, 3].map { it + 1 }.sum # map is redundant here, but gets no notice
[2, 3].sum { it + 1 } # returns same result
 
```

**Comment by ydakuka** [comment]
```ruby
obj.to_a.to_h
```
It doesn’t really make sense, so there is no such cop. It’s not related to `#each_slice(i)` or any other method.

**Comment by forthrin** [comment]
What about map/sum vs sum? Isn't this a redundancy that should be caught?

**Comment by ydakuka** [comment]
This is already covered by `Performance/Sum` from https://github.com/rubocop/rubocop-performance:
https://docs.rubocop.org/rubocop-performance/cops_performance.html#performancesum

```ruby
$ rubocop file.rb
Inspecting 1 file
W

Offenses:

file.rb:7:12: C: [Correctable] Performance/Sum: Use sum { ... } instead of map { ... }.sum.
    [2, 3].map { it + 1 }.sum
           ^^^^^^^^^^^^^^^^^^
```

**Comment by forthrin** [comment]
Ah. It's not enabled by default.

What are the **command line switches** for:

1. Enable all cops in a plugin
2. Enable NewCops

Also:

3. Why is TargetRubyVersion 2.7 by default, and not `$(which ruby) --version`. Is this a planned change?

**Comment by ydakuka** [comment]
```yaml
AllCops:
  EnabledByDefault: true
  NewCops: enable
```
https://docs.rubocop.org/rubocop/configuration.html#enabled

https://docs.rubocop.org/rubocop/extensions.html

https://docs.rubocop.org/rubocop/compatibility.html

**Comment by forthrin** [comment]
Asked not for config file statements, but CLI switches (eg. `rubocop --enable-all-cops rubocop-performance`)

Can't find how to enable all cops for given plugin in extensions link.

The compatibility link mentions *dropping* 2.7 support, not why it's the default.

**Comment by ydakuka** [comment]
You just need to install gems and add them as plugins. After that, you can use them as shown in the example https://github.com/rubocop/rubocop/issues/14640#issuecomment-3679006303

**Comment by forthrin** [comment]
Sorry, but you're answering something else than asked. I think I might post the underlying goal of the questions as a feature request to ease bug reporting.
